python/openai/tests/test_tool_calling.py

```diff
@@ -0,0 +1,571 @@
+        assert streamed_args.get("city") == "Dallas"
+        assert streamed_args.get("state") == "TX"
+
+        # make sure everything matches non-streaming except for ID
+        assert function_name == tool_calls[0].function.name
+        assert choice.message.role == role_name
+        assert choice.message.tool_calls[0].function.name == function_name
+
+        # compare streamed with non-streamed args Dict-wise, not string-wise
+        # because character-to-character comparison might not work e.g. the tool
+        # call parser adding extra spaces or something like that. we care about the
+        # dicts matching not byte-wise match
+        assert parsed_arguments == streamed_args
+
+    @pytest.mark.asyncio
+    async def test_tool_call_with_reply_response(
+        self, client: openai.AsyncOpenAI, model: str, backend: str
+    ):
+        chat_completion = await client.chat.completions.create(
+            messages=MESSAGES_WITH_TOOL_RESPONSE,
+            temperature=0,
+            max_tokens=128,
+            model=model,
+            tools=[WEATHER_TOOL, WEATHER_FORECAST_TOOL],
+            logprobs=False,
+        )
+
+        choice = chat_completion.choices[0]
+
+        assert choice.finish_reason != "tool_calls"  # "stop"
+        assert choice.message.role == "assistant"
+        assert choice.message.tool_calls is None or len(choice.message.tool_calls) == 0
+        assert choice.message.content is not None
+
+        stream = await client.chat.completions.create(
+            messages=MESSAGES_WITH_TOOL_RESPONSE,
+            temperature=0,
+            max_tokens=128,
+            model=model,
+            tools=[WEATHER_TOOL, WEATHER_FORECAST_TOOL],
+            logprobs=False,
+            stream=True,
+        )
+
+        chunks: List[str] = []
+        finish_reason_count = 0
+        role_sent: bool = False
+
+        async for chunk in stream:
+            delta = chunk.choices[0].delta
+
+            if delta.role:
+                assert not role_sent
+                assert delta.role == "assistant"
+                role_sent = True
+
+            if delta.content:
+                chunks.append(delta.content)
+
+            if chunk.choices[0].finish_reason is not None:
+                finish_reason_count += 1
+                assert chunk.choices[0].finish_reason == choice.finish_reason
+
+            assert not delta.tool_calls or len(delta.tool_calls) == 0
+
+        assert role_sent
+        assert finish_reason_count == 1
+        assert len(chunks)
+
+        # FIXME: tensorrt_llm streaming and non-streaming requests are not generating exact same text response in this case
+        # example:
+        #   streaming: The current temperature in Dallas, TX is 98°F. Is there anything else I can assist you with?
+        #   non-streaming: It's currently 98°F in Dallas.
+        if backend != "tensorrtllm":
+            assert "".join(chunks) == choice.message.content
```

Comment:
Can you instead do something like this so we can automate detecting when TRTLLM fixes/changes this behavior?
```python
if backend == "tensorrtllm":
  # FIXME: tensorrt_llm streaming and non-streaming requests are not generating exact same text response in this case
  # NOTE: If this assertion consistently does not get raised, then this means the TRTLLM behavior 
  # is changed and this special case can be removed.
  with pytest.raises ... # or pytest.xfail or something
    assert "".join(chunks) == choice.message.content
else:
  assert "".join(chunks) == choice.message.content